RFS/src/main/java/com/rfs/cms/LeaseExpireTrigger.java

```diff
@@ -39,11 +39,20 @@ public LeaseExpireTrigger(Consumer<String> onLeaseExpired, Clock currentTimeSupp
     public void registerExpiration(String workItemId, Instant killTime) {
         workItemToLeaseMap.put(workItemId, killTime);
         final var killDuration = Duration.between(currentTimeSupplier.instant(), killTime);
-        scheduledExecutorService.schedule(() -> {
+        final Runnable exirationRunnable = () -> {
             if (workItemToLeaseMap.containsKey(workItemId)) {
                 onLeaseExpired.accept(workItemId);
             }
-        }, killDuration.toMillis(), TimeUnit.MILLISECONDS);
+        };
+        if (killDuration.isNegative()) {
+            exirationRunnable.run();
```

Comment:
Curious - could you do this w/ a parameter of 0 or 1 ms from now and tighten the code (not have to create a lambda w/out a context)

Reply:
I'm not sure what you mean by this